linker/Mono.Linker.Steps/SweepStep.cs

```diff
@@ -175,6 +175,8 @@ bool ResolveAllTypeReferences (AssemblyDefinition assembly)
 			foreach (TypeReference tr in assembly.MainModule.GetTypeReferences ()) {
 				if (hash.ContainsKey (tr))
 					continue;
+				if (!CanChangeScope (tr))
```

Comment:
If you don't want to change the scope you should move all the logic to the virtual method and override it with no-op